CMake/External_qtExtensions.cmake

```diff
@@ -1,14 +1,27 @@
-# Make sure we're using Qt 4
-if(NOT (Qt_SELECT_VERSION VERSION_LESS 5 AND NOT (Qt_SELECT_VERSION VERSION_LESS 4)))
-  message(FATAL_ERROR "qtExtensions requires Qt major version 4")
-endif()
-
 # Qt
-add_package_dependency(
-  PACKAGE qtExtensions
-  PACKAGE_DEPENDENCY Qt
-  PACKAGE_DEPENDENCY_ALIAS Qt4
-)
+
+if(Qt_version_major EQUAL 4)
+  add_package_dependency(
+    PACKAGE qtExtensions
+    PACKAGE_DEPENDENCY Qt
+    PACKAGE_DEPENDENCY_ALIAS Qt4
+  )
+  set(QT_ARGS
+    -DQT_QMAKE_EXECUTABLE:PATH=${QT_QMAKE_EXECUTABLE}
+    -DQTE_QT_VERSION:STRING=4
+  )
+else()
+  add_package_dependency(
+    PACKAGE qtExtensions
+    PACKAGE_DEPENDENCY Qt
+    PACKAGE_DEPENDENCY_ALIAS Qt5
+    PACKAGE_DEPENDENCY_COMPONENTS Widgets Xml
```

Comment:
qtExtensions links to other components as well like UiPlugin, etc.
https://github.com/Kitware/qtextensions/blob/master/CMakeLists.txt#L309-L313

Reply:
...but they're optional(ish). OTOH maybe we want to force that we are able to build all of the optional stuff?

Reply:
That's right but I assumed the designer plugin is ON by default and hence we need those modules.

Reply:
I think, yes, but the relevant question is if we'd rather fletch build "as much as possible" or "as minimal as possible". Since the consensus seems to be "as much as possible", I'll change this.